Sources/SourceControl/RepositoryManager.swift

```diff
@@ -230,10 +233,20 @@ public class RepositoryManager {
     /// - Throws:
     /// - Returns: Details about the performed fetch.
     @discardableResult
-    func fetchAndPopulateCache(handle: RepositoryHandle, repositoryPath: AbsolutePath) throws -> FetchDetails {
+    func fetchAndPopulateCache(handle: RepositoryHandle, repositoryPath: AbsolutePath, on queue: DispatchQueue) throws -> FetchDetails {
         var cacheUsed = false
         var cacheUpdated = false
 
+        func updateFetchProgress(progress: FetchProgress) -> Void {
+            queue.async {
+                if let total = progress.totalSteps {
```

Comment:
flip line 241 and 242 to save redundant queue operations

Reply:
done cdac447